Shopify/cli
# Make it so we load the app from config when possible

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/645
Fixes https://github.com/Shopify/internal-cli-foundations/issues/694

### WHAT is this pull request doing?

Don't ask the user to select an app if they have opted in config-as-code.

### How to test your changes?

- `bin/create-test-app.js -e ui --config-as-code`
- `pnpm shopify app env show --path ~/Desktop/nightly-app-something` (you shouldn't see the app prompt)

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
